types/hashing.go

```diff
@@ -0,0 +1,24 @@
+package types
+
+import (
+	"encoding"
+
+	"github.com/minio/sha256-simd"
+)
+
+func (h *Header) Hash() [32]byte {
+	return hash(h)
+}
+
+func (b *Block) Hash() [32]byte {
```

Comment:
Question: What's the rationale to use an array and not slice?

Reply:
- `crypto/sha3` returns array
- we used arrays in structs, because hash is fixed size (logically)

Reply:
> we used arrays in structs, because hash is fixed size (logically)
Totally, but in practice, we don't use fixed-size arrays and kinda forcing users to always slice it themselves, but that is ok as long you ok with that.